File: vault-door3/flag.py
s = "jU5t_a_sna_3lpm18gb41_u_4_mfr340"
lst = []

for a in range(8):
      lst.insert(a, s[a])

for b in range(8,16):
      lst.insert(b, s[23 - b])

for c in range(16, 32, 2):
      lst.insert(c, s[46 - c])

for d in range(17, 32, 2):
      lst.insert(d, s[d])

flag = 'picoCTF{'
for v in lst:
      flag += v

print(flag + '}')

File: vault-door3/flag.js
password = "jU5t_a_sna_3lpm18gb41_u_4_mfr340"
// password = "jU5t_a_sna_3lpm1dg347_u_4_mfr54b"
var i;
var buffer = Array(32);

for (i = 0; i < 8; i++) {
      buffer[i] = password.charAt(i)
}

for (; i < 16; i++) {
      buffer[i] = password.charAt(23 - i)
}
for (; i < 32; i += 2) {
      buffer[i] = password.charAt(46 - i)
}
for (i = 31; i >= 17; i -= 2) {
      buffer[i] = password.charAt(i)
}

console.log("picoCTF{" + buffer.join("") + "}");